src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidator.java

```diff
@@ -1,60 +1,117 @@
 package uk.ac.ebi.subs.filecontentvalidator.service;
 
-import uk.ac.ebi.subs.data.fileupload.FileStatus;
-import uk.ac.ebi.subs.filecontentvalidator.exception.ErrorMessages;
+import lombok.Data;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.ac.ebi.subs.filecontentvalidator.config.CommandLineParams;
 import uk.ac.ebi.subs.filecontentvalidator.exception.FileNotFoundException;
-import uk.ac.ebi.subs.repository.model.fileupload.File;
-import uk.ac.ebi.subs.repository.repos.fileupload.FileRepository;
+import uk.ac.ebi.subs.filecontentvalidator.exception.NotSupportedFileTypeException;
+import uk.ac.ebi.subs.validator.data.SingleValidationResult;
+import uk.ac.ebi.subs.validator.data.structures.SingleValidationResultStatus;
+import uk.ac.ebi.subs.validator.data.structures.ValidationAuthor;
 
-import java.util.Arrays;
-import java.util.List;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.UUID;
 
+@Data
+@Service
 public class FileContentValidator {
```

Comment:
We may as well call this class what it is - a fastq content validator

Reply:
Yes for now, but I wanted to make it generic

Reply:
How about one implementation class per format?

Reply:
never mind, we can leave this till there's more than one type to validate